zerver/lib/streams.py

```diff
@@ -568,7 +569,10 @@ def user_has_content_access(
     if user_profile.is_guest:
         return False
 
-    if stream.is_public():
+    if not exclude_zephyr_realms_from_public_check and stream.is_public():
+        return True
+
+    if exclude_zephyr_realms_from_public_check and not stream.invite_only:
         return True
```

Comment:
Better to do these two checks in the opposite order -- then the first check can remain just `if stream.is_public()`.

Reply:
``` diff
diff --git a/zerver/lib/streams.py b/zerver/lib/streams.py
index 35f57e20d8..493a6cce24 100644
--- a/zerver/lib/streams.py
+++ b/zerver/lib/streams.py
@@ -571,10 +571,10 @@ def user_has_content_access(
     if user_profile.is_guest:
         return False
 
-    if not exclude_zephyr_realms_from_public_check and stream.is_public():
+    if exclude_zephyr_realms_from_public_check and not stream.invite_only:
         return True
 
-    if exclude_zephyr_realms_from_public_check and not stream.invite_only:
+    if stream.is_public():
         return True
 
     if user_group_membership_details.user_recursive_group_ids is None:
```